src/backend/distributed/utils/multi_transaction.c

```diff
@@ -347,3 +348,62 @@ ConnectionList(HTAB *connectionHash)
 
 	return connectionList;
 }
+
+
+/*
+ * OpenConnectionsToShardPlacements opens connections to all placements of the
+ * shard with the given shardId and populates the shardConnectionHash table
+ * accordingly.
+ */
+void
+OpenConnectionsToShardPlacements(uint64 shardId, HTAB *shardConnectionHash,
+								 char *nodeUser)
+{
+	bool shardConnectionsFound = false;
+
+	/* get existing connections to the shard placements, if any */
+	ShardConnections *shardConnections = GetShardConnections(shardConnectionHash,
+															 shardId,
+															 &shardConnectionsFound);
+
+	List *shardPlacementList = FinalizedShardPlacementList(shardId);
```

Comment:
It seems like we never lock the metadata for the shards. Not this PR's fault, but this could cause things to break in the event that shard rebalancer runs during a DDL command.